polkadot/xcm/xcm-builder/src/controller.rs

```diff
@@ -51,13 +53,15 @@ pub trait ExecuteControllerWeightInfo {
 /// - Validating and Converting the origin to a Location.
 /// - Handling versioning.
 /// - Calling  the internal executor, which implements [`ExecuteXcm`].
-pub trait ExecuteController<Origin, RuntimeCall> {
+pub trait ExecuteController<Origin, RuntimeCall, MaxXcmEncodedSize> {
 	/// Weight information for ExecuteController functions.
 	type WeightInfo: ExecuteControllerWeightInfo;
 
 	/// Attempt to execute an XCM locally, returns Ok with the weight consumed if the execution
 	/// complete successfully, Err otherwise.
 	///
+	/// WARNING: This call is DEPRECATED! Use `execute_blob` instead.
+	///
```

Comment:
maybe just use a `#[deprecated*note = "..."]` attribute instead of these WARNING?

Reply:
I put it in the doc comment since those get added to the metadata, and UIs won't see the rust deprecation warnings but they will see the metadata.
I could add two, one for the UIs and one for internal use